src/gui/FileBrowser.cpp

```diff
@@ -326,6 +328,49 @@ FileBrowserTreeWidget::FileBrowserTreeWidget(QWidget * parent ) :
 
 }
 
+QList<QString> FileBrowserTreeWidget::expandedDirs() const
+{
+	QList<QString> expandedDirs;
+	for( int i = 0; i < topLevelItemCount(); ++i )
+	{
+		QTreeWidgetItem * item  = topLevelItem( i );
+
+		// Add expanded top level directories.
+		if (item->isExpanded() && (item->type() == TypeDirectoryItem))
+		{
+			Directory *d = static_cast<Directory *> ( item );
+			expandedDirs.append( d->fullName() );
+		}
+
+		// Add expanded child directories.
+		expandedDirs.append(
+			expandedChildDirs( item )
+		);
+
+	}
+	return expandedDirs;
+}
+
+QList<QString> FileBrowserTreeWidget::expandedChildDirs( QTreeWidgetItem * item ) const
```

Comment:
While both functions look correct to me, you could turn both into one, like in `FileBrowser::expandItems` with the `?:`-operator. However, this is just a suggestion.

Reply:
Yes I like that :-D